modules/user/controller.js

```diff
@@ -0,0 +1,224 @@
+        status: "error",
+        code: 401,
+        message: "Unauthorized",
+        data: "Unauthorized",
+      });
+    }
+    req.user = user;
+
+    next();
+  })(req, res, next);
+};
+
+export const login = async (req, res, next) => {
+  const { email, password } = req.body;
+  const schema = Joi.object({
+    email: Joi.string().email().required(),
+    password: Joi.string()
+      .pattern(new RegExp("^[a-zA-Z0-9]{3,30}$"))
+      .required(),
+  });
+  const { error } = schema.validate({
+    email: email,
+    password: password,
+  });
+
+  if (error) return res.status(400).json(error.details[0].message);
+  try {
+    const user = await userService.findUserByEmail(email);
+    if (!user || !user.validPassword(password)) {
+      return res.status(400).json({
+        status: "error",
+        code: 400,
+        message: "Incorrect login or password",
+        data: "Bad request",
+      });
+    }
+
+    const payload = {
+      id: user.id,
+      useremail: user.email,
+    };
+
+    const token = jwt.sign(payload, secret, { expiresIn: "1h" });
+    await userService.saveToken(user.id, token);
+    res.json({
+      status: "success",
+      code: 200,
+      data: {
+        token,
+        user: {
+          email: user.email,
+          subscription: user.subscription,
+        },
+      },
+    });
+  } catch (e) {
+    console.error(e);
+    next(e);
+  }
+};
+
+export const signup = async (req, res, next) => {
+  const { email, password } = req.body;
+  const schema = Joi.object({
+    email: Joi.string().email().required(),
+    password: Joi.string()
+      .pattern(new RegExp("^[a-zA-Z0-9]{3,30}$"))
+      .required(),
+  });
+  const { error } = schema.validate({
+    email: email,
+    password: password,
+  });
+
+  if (error) return res.status(400).json(error.details[0].message);
+
+  try {
+    const user = await userService.findUserByEmail(email);
+    if (user) {
+      return res.status(409).json({
+        status: "error",
+        code: 409,
+        message: "Email is already in use",
+        data: "Conflict",
+      });
+    }
+    const avatarURL = gravatar.url(email);
+    const newUser = await userService.register(email, password, avatarURL);
+    const {
+      email: emailRegistered,
+      subscription: subscriptionRegistered,
+      avatarURL: avatarURLRegistered,
+    } = newUser;
+
+    res.status(201).json({
+      status: "success",
+      code: 201,
+      data: {
+        user: {
+          email: emailRegistered,
+          subscription: subscriptionRegistered,
+          avatarURL: avatarURLRegistered,
+        },
+      },
+    });
+  } catch (error) {
+    next(error);
+  }
+};
+export const logout = async (req, res, next) => {
+  const id = req.user;
+
+  try {
+    await userService.saveToken(id, null);
+
+    res.json({
+      status: "success",
+      code: 204,
+    });
+  } catch (e) {
+    console.error(e);
+    next(e);
+  }
+};
+export const current = async (req, res, next) => {
```

Comment:
Tu masz już usera w req nie musisz go ponownie pobierać i sprawdzać, wystarczy sam to:

```
export const current = async (req, res, next) => {
    return res.json({
      status: "success",
      code: 200,
      data: {
        user: {
          email: req.user.email,
          subscription: req.user.subscription,
        },
      },
    });
}
```